src/base/utils/net.cpp

```diff
@@ -83,6 +100,73 @@ namespace Utils
             return subnet.first.toString() + u'/' + QString::number(subnet.second);
         }
 
+        std::pair<QHostAddress, QHostAddress> subnetToIpRange(const Subnet &subnet)
+        {
+            const QHostAddress &address = subnet.first;
+            int prefixLength = subnet.second;
+
+            auto addressFamily = address.protocol();
+
+            if (addressFamily == QAbstractSocket::IPv4Protocol)
+            {
+                quint32 ip = address.toIPv4Address();
+                quint32 mask = 0;
+
+                if (prefixLength >= 0 && prefixLength <= 32)
+                {
+                    mask = (0xFFFFFFFF << (32 - prefixLength)) & 0xFFFFFFFF;
+                }
+
+                quint32 network = ip & mask;
+                quint32 broadcast = network | (~mask & 0xFFFFFFFF);
+
+                QHostAddress start(network);
+                QHostAddress end(broadcast);
+
+                return std::make_pair(start, end);
+            }
+            else if (addressFamily == QAbstractSocket::IPv6Protocol)
+            {
+                quint8 ip6[16];
+                quint8 mask6[16] = {0};
+
+                QIPv6Address addressBytes = address.toIPv6Address();
```

Comment:
According to Qt documentation, QIPv6Address is not supposed to be used directly, but as Q_IPV6ADDR.